connect/tls.go

```diff
@@ -273,6 +297,9 @@ func newDynamicTLSConfig(base *tls.Config) *dynamicTLSConfig {
 	}
 	if len(base.Certificates) > 0 {
 		cfg.leaf = &base.Certificates[0]
+		// If this does error then future calls to Ready will fail
+		// It is better to handle not-Ready rather than failing
```

Comment:
I see what you mean and kinda agree we shouldn't fail hard on construction, however if this happens it's likely that the instnce will never get healthy and it won't be at all clear why since this error is swallowed.

I *think* the only time we actually have a certificate at construction time is when we are in Dev mode with static certs (i.e. never) so it's somewhat moot, but I suggest that we log the error here and then continue so at least if we do ever use this code path it's not silent on errors.